src/components/Button/ButtonContent/index.test.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { ButtonContent, layoutType } from '.'
+import renderWithTheme from '../../../testing/renderWithTheme'
+import mentor from '../../../assets/mentor.svg'
+
+test('renders button content with layout center', () => {
+	const { baseElement } = renderWithTheme(
+		<ButtonContent layout={layoutType.center} />
+	)
+
+	expect(baseElement).toMatchInlineSnapshot(`
+		<body>
+		  <div>
+		    <div
+		      class="sc-hKgILt eNcJhZ"
+		    />
+		  </div>
+		</body>
+	`)
+})
+
+test('renders button content with layout line', () => {
+	const { baseElement } = renderWithTheme(
+		<ButtonContent layout={layoutType.line} />
+	)
+
+	expect(baseElement).toMatchInlineSnapshot(`
+		<body>
+		  <div>
+		    <div
+		      class="sc-eCssSg eOWPJF"
+		    />
+		  </div>
+		</body>
+	`)
+})
+
+// @NOTE: We don't know how to test the image
+test('renders button content with layout center and alt', () => {
+	const { getByAltText } = renderWithTheme(
+		<ButtonContent layout={layoutType.center} icon={mentor} alt="Mentor" />
+	)
+
+	const imageElement = getByAltText('Mentor')
+	expect(imageElement.tagName).toBe('IMG')
+})
+
+test('renders button content with layout center and title', () => {
+	const { getByText } = renderWithTheme(
+		<ButtonContent layout={layoutType.center} title="Mentor" />
+	)
+
+	const spanElement = getByText('Mentor')
+	expect(spanElement.tagName).toBe('SPAN')
+})
+
+test('renders button content with layout center and subtitle', () => {
+	const { getByText } = renderWithTheme(
+		<ButtonContent
+			layout={layoutType.center}
+			subtitle="Poznej nové přátele."
+		/>
+	)
+
+	const pElement = getByText('Poznej nové přátele.')
+	expect(pElement.tagName).toBe('P')
+})
+
+test('renders button content with layout center, icon, alt, title and subtitle', () => {
+	const { baseElement } = renderWithTheme(
+		<ButtonContent
+			layout={layoutType.center}
+			icon={mentor}
+			alt="Mentor"
+			title="Mentee"
+			subtitle="Poznej nové přátele."
+		/>
+	)
+
+	expect(baseElement).toMatchInlineSnapshot(`
+		<body>
```

Comment:
Ještě těm testům moc nerozumí, ale musí být takhle určená celá struktura? Hledání class v img v divu v jiném divu v body mi přijde takové složité. Kdybychom náhodou přidali nějaký div, tak už nám to neprojde? 

Pokud bychom udělali ikonky přes nějakou tu knihovnu, test obrázku nebude vůbec potřeba. :thinking:

Reply:
No pokud bychom něco přidali nebo odebrali, tak by nás měl právě test upozornit na změnu (chybu) a pokud bychom tu změnu, tak chtěli, musí se test přepsat. V tomhle případě smazat obsah v závorkách a nechat vygenerovat nový.